WordPress/wordpress-playground
# wp now: load virtually config constants to allow running multiple projects at the same time

- Closes https://github.com/WordPress/wordpress-playground/issues/324

## What?



- New blueprint function to autoload virtually WP config contains. It doesn't write or update user files, it only creates a file in VFS and use php.ini [`auto_prepend_file`](https://www.php.net/manual/en/ini.core.php#ini.auto-prepend-file)
- It allows to run multiple `wp-now` projects at the same time
- It includes tests.

## Why?



The current blueprint `defineSiteUrl` creates a `json` file that is re-used in every wp-now instance. Creating conflicts between site redirections. See https://github.com/WordPress/wordpress-playground/issues/324

## How?



- Created `defineVirtualWpConfigConsts` an alternative of `defineWpConfigConsts`.

## Testing Instructions






* Check out this branch.
* Prepare two directories with a theme or plugin
* run `npx nx preview wp-now start --path=/path/to/theme-plugin-1`
* Observe it the WordPress environment works as expected
* run `npx nx preview wp-now start --path=/path/to/theme-plugin-2`
* Observe the new instance runs as expected
* Observe the first instance is still working in a different port without any URL redirection.
* Observe the `wp-config.php` was not modified and you don't have any new `playground-consts.json.json` file in the root of your folder. (For a clean test, remove your files inside  `~/.wp-now`)
